src/train-classifier/src/main/scala/TrainClassifier.scala

```diff
@@ -36,9 +37,13 @@ class TrainClassifier(override val uid: String) extends Estimator[TrainedClassif
   def getNumFeatures: Int = $(numFeatures)
   def setNumFeatures(value: Int): this.type = set(numFeatures, value)
 
-  val indexLabel = BooleanParam(this, "indexLabel", "index the label column", true)
-  def getIndexLabel: Boolean = $(indexLabel)
-  def setIndexLabel(value: Boolean): this.type = set(indexLabel, value)
+  val reindexLabel = BooleanParam(this, "reindexLabel", "reindex the label column", true)
+  def getReindexLabel: Boolean = $(reindexLabel)
```

Comment:
added comments to the parameter and to the class to explain the interaction between the reindex labels and labels parameters